circuitbreaker/circuitbreaker.go

```diff
@@ -10,14 +10,30 @@ import (
 	"github.com/slok/goresilience/metrics"
 )
 
-type state string
+type state uint
 
 const (
-	stateOpen     state = "open"
-	stateHalfOpen state = "halfopen"
-	stateClosed   state = "closed"
+	stateNew   state = iota
+	stateOpen
+	stateHalfOpen
+	stateClosed
 )
 
+var stateStrings = []string {
+	"new",
+	"open",
+	"halfopen",
+	"closed",
+}
+
+func (st state) label () string {
+	return stateStrings[st]
```

Comment:
2 things:

- Instead of `label` what do you think about implementing the [Stringer](https://golang.org/pkg/fmt/#Stringer) interface?

- This could panic if a state does not have an associated string in the array, although we control this it could lead to a runtime panic in a change. A better approach could be using a `switch` with a fallback string, would be safer.

Reply:
I'll have to read up on the Stringer interface but it feels a more natural fit.
I agree using a ```switch``` offers a more defensive implementation.

Will consider both when refactoring.